src/arrayMethodSort.js

```diff
@@ -4,8 +4,24 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const n = this.length;
+
+    for (let i = 0; i < n - 1; i++) {
+      for (let j = 0; j < n - 1 - i; j++) {
+        if (compareFunction) {
+          if (compareFunction(this[j], this[j + 1]) > 0) {
+            [this[j], this[j + 1]] = [this[j + 1], this[j]];
+          }
+        } else {
+          if (String(this[j]) > String(this[j + 1])) {
+            [this[j], this[j + 1]] = [this[j + 1], this[j]];
+          }
+        }
+      }
+    }
+
+    return this;
```

Comment:
The bubble sort algorithm is correctly implemented, but it is not the most efficient sorting algorithm for larger datasets. If performance is a concern, consider researching and implementing more efficient algorithms like quicksort or mergesort.